### PR TITLE
William/upgrades

### PR DESCRIPTION
**Pull request by swansontec**
This branch starts off by upgrading everything in package.json, then fixing the fallout. Finally, it adds support for React. I have cleaned this up compared to when I first did it back in March, and I think it's ready for testing & review.

I might want to try these rules out in react-native-patina and react-native-patina to see how they work in a real-world setting.

### CODE REVIEW

**Comment by samholmes** [comment]
I was able to test this branch with the New Sync Server (https://github.com/EdgeApp/edge-sync-server). My only comments are that the `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` dependencies that the host package/project has installed needs to version match what is included in `eslint-config-standard-kit`. This branch includes version `^4.8.2` for both deps, but the latest version as of writing is `^4.9.1`. Perhaps peerDependencies could be useful here?

**Comment by swansontec** [comment]
If our rules need version `^4.8.2`, a version like `^4.9.1` should be compatible, right? The problem with `peerDependencies` is that putting in a rule for something like `@typescript-eslint/parser` applies to everyone, even people who don't use Typescript, so they will get weird ugly errors if they don't have it installed.

**Comment by samholmes** [comment]
@swansontec I see how that issue with `peerDependencies` would be a show-stopper. It would be annoying for users of `eslint-config-standard-kit` to be required to install all peer dependencies even when they don't plan to use say typescript or JSX for example. Though, by leaving dependencies and not changing them to peerDependencies, they are still being installed for the user... I wonder if there is a way to get best of both worlds (install some compatible version for the user, but also maintain the peer dep relationship).

As far was the issue I was seeing with the two versions, it was due to a mis-use of updot when testing out the dep versions. What must had happened was when updating the deps to `4.9.1`, another run of `yarn updot eslint-config-standard-kit` was required to keep the WIP branch in use.

File: src/rules/rule-files.js [hunk 1]
```diff
@@ -124,19 +132,33 @@ function makeFiles() {
...
-          parser: '@typescript-eslint/parser',
-          ...typescriptConfig.overrides[0]
+          ...typescriptConfig.overrides[0],
+          rules: {
+            ...removeProps(typescriptConfig.overrides[0].rules, [
+              // This feature requires a polyfill,
+              // which we shouldn't force on people:
+              '@typescript-eslint/prefer-includes',
+              // These rules require too much manual effort to satisfy:
+              '@typescript-eslint/prefer-nullish-coalescing',
```
Comment (swansontec) [inline src/rules/rule-files.js:150]: Actually, leave these rules in place. We can disable on a per-project basis.

**Review by samholmes — APPROVED** [review]
All good. After making the minor change, we're good to go

**Review by swansontec — COMMENTED** [review]
One little change is needed before merging.

**Pull request merged**